sympy/solvers/ode/single.py

```diff
@@ -784,5 +784,72 @@ def _get_general_solution(self, *, simplify: bool = True):
         return [gensol]
 
 
+class SecondNonlinearAutonomousConserved(SinglePatternODESolver):
+    r"""
+    Gives solution for the autonomous second order nonlinear
+    differential equation of the form
+
+    .. math :: f''(x) = g(f(x))
+
+    The solution for this differential equation can be computed
+    by multiplying by `f'(x)` and integrating on both sides,
+    converting it into a first order differential equation.
+
+    Examples
+    ========
+
+    >>> from sympy import Function, symbols, dsolve
+    >>> f, g = symbols('f g', cls=Function)
+    >>> x = symbols('x')
+
+    >>> eq = f(x).diff(x, 2) - g(f(x))
+    >>> dsolve(eq, simplify=False)
+    [Eq(Integral(1/sqrt(C1 + 2*Integral(g(_u), _u)), (_u, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(C1 + 2*Integral(g(_u), _u)), (_u, f(x))), C2 - x)]
+
+    >>> from sympy import exp, log
+    >>> eq = f(x).diff(x, 2) - exp(f(x)) + log(f(x))
+    >>> dsolve(eq, simplify=False)
+    [Eq(Integral(1/sqrt(-2*_u*log(_u) + 2*_u + C1 + 2*exp(_u)), (_u, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(-2*_u*log(_u) + 2*_u + C1 + 2*exp(_u)), (_u, f(x))), C2 - x)]
+
+    References
+    ==========
+
+    http://eqworld.ipmnet.ru/en/solutions/ode/ode0301.pdf
+    """
+    hint = "2nd_nonlinear_autonomous_conserved"
+    has_integral = True
+    order = [2]
+
+    def _wilds(self, f, x, order):
+        fy = Wild('fy', exclude=[0, f(x).diff(x), f(x).diff(x, 2)])
+        return (fy,)
+
+    def _equation(self, fx, x, order):
+        fy = self.wilds()[0]
+        return fx.diff(x, 2) + fy
+
+    def _verify(self, fx):
+        fy = self.wilds_match()[0]
+        u = Dummy('u')
+        x = self.ode_problem.sym
+        syms = fy.subs(fx, u).free_symbols
+        if x in syms:
```

Comment:
We should probably add this as a `is_autonomous` method on `ode_problem`.

Reply:
Ah, yes, that is a nice idea. Would be useful later on.

Reply:
Should I add `is_autonomous` to `SingleODEProblem` or `SinglePatternODESolver`? If it is the former, how would I be able to call the `wilds_match` as it wouldn't be defined?

Reply:
I think it should be for `SingleODEProblem` as being autonomous is a property of the problem rather than a property of any solver.

Rather than using `wilds_match` we should think of a way to test for a generic ODE whether or not it is autonomous. I think that the technique that you use with `subs` here would probably work in general e.g.:
```python
In [7]: eq1 = f(x).diff(x, 2) + exp(f(x)) * f(x).diff(x)

In [8]: eq1.subs(f(x), y)
Out[8]: 
             2   
 y d        d    
ℯ ⋅──(y) + ───(y)
   dx        2   
           dx    

In [9]: x in eq1.subs(f(x), y).free_symbols
Out[9]: False

In [10]: eq2 = f(x).diff(x) + x*f(x)

In [11]: x in eq2.subs(f(x), y).free_symbols
Out[11]: True
```

Reply:
I've done this. Should I add any tests for this? If so, where do you think they should go?

Reply:
Are there already some tests for SingleODEProblem somewhere?

Reply:
Yes, found them. I had overlooked them earlier.